### ISSUE TITLE
Expression YEAR method does not exist.

### ISSUE THREAD

**Issue by progelio** [mbdavid/LiteDB#1045]
I am trying to create an index using an expression but it says "YEAR method does not exist".

db.collection.EnsureIndex("CreatedDate", "DATE(YEAR($.CreatedDate), MONTH($.CreatedDate), DAY($.CreatedDate))", false);

The "CreatedDate" property is a DateTime?, nullable.

My goal is to index only the date portion and drop the time.

**Comment by mbdavid** [comment]
Hi @progelio, I made this example and works as expected... (i'm using 4.1.4)

```C#
var doc = new BsonDocument { ["CreatedDate"] = DateTime.Now };

var e = new BsonExpression("DATE(YEAR($.CreatedDate), MONTH($.CreatedDate), DAY($.CreatedDate))");

Console.WriteLine(e.Execute(doc).First().AsDateTime);
```

Do you have stack trace so I can track where this happend

**Comment by progelio** [comment]
Confirmed. I didn't realize that the YEAR function was not available in version 4.0. I have now updated to the latest version. Thanks!
